src/BreakpointViewer.cpp

```diff
@@ -0,0 +1,621 @@
+#include <QPainter>
+#include <QPaintEvent>
+#include <QMessageBox>
+#include <QToolTip>
+#include <QComboBox>
+#include <QDebug>
+#include <cassert>
+#include "BreakpointViewer.h"
+#include "ui_BreakpointViewer.h"
+#include "Convert.h"
+#include "CommClient.h"
+#include "OpenMSXConnection.h"
```

Comment:
Include order:
We prefer to order the #includes from "near" to "far". That is the #include for the header corresponding to this .cpp file first (BreakPointViewer.h"). Followed by other debugger header files, followed by Qt headers, followed by c++ headers. (And possibly alphabetically sorted within a level).

Rationale: this order makes it more likely to detect errors against the rule/guideline that "headers should be self-contained", that is a header should #includes everything that it needs.

Note (this applies to all my review comments):
Probably this rule isn't fully followed in all current debugger code. But when writing new code it's a good idea to immediately try to use the general "openMSX" style.